src/PipeSpec.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Mvc\Middleware;
+
+use Closure;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Webmozart\Assert\Assert;
+
+final class PipeSpec
+{
+    /**
+     * @var array
+     */
+    private $spec;
+
+    /**
+     * @param string|RequestHandlerInterface|MiddlewareInterface|Closure ...$spec
+     */
+    public function __construct(...$spec)
+    {
+        $this->spec = $spec;
+    }
```

Comment:
I would like to assert value types here but since it will be in the config and cached config I am somewhat cautious.

Reply:
With a [type hint](https://github.com/laminas/laminas-mvc-middleware/pull/3#discussion_r526120010), at least static analyzers will catch a wrong type :)